sim/mage/arcane/arcane.go

```diff
@@ -80,12 +79,28 @@ func (arcaneMage *ArcaneMage) ApplyArcaneSpecInnate() {
 		Label: "Mana Adept",
 		//ActionID: core.ActionID{SpellID: 76547},
 		OnGain: func(aura *core.Aura, sim *core.Simulation) {
-			arcaneMastery.UpdateFloatValue(arcaneMage.CurrentMana() / arcaneMage.MaxMana() * arcaneMage.GetArcaneMasteryBonus())
+			arcaneMastery.UpdateFloatValue(arcaneMage.ArcaneMasteryValue())
 			arcaneMastery.Activate()
 		},
 		OnExpire: func(aura *core.Aura, sim *core.Simulation) {
 			arcaneMastery.Deactivate()
 		},
 	}))
 
+	core.MakeProcTriggerAura(&arcaneMage.Unit, core.ProcTrigger{
+		Name:     "Arcane Mastery Mana Updater",
+		Callback: core.CallbackOnCastComplete,
+		Handler: func(sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+			arcaneMastery.UpdateFloatValue(arcaneMastery.GetFloatValue())
```

Comment:
I think you copy/pasted the wrong func here and want to use `arcaneMage.ArcaneMasteryValue()`

Reply:
oops 🙃fixed